docs/release.md

```diff
@@ -47,6 +47,44 @@ git push upstream ${RELEASE_TAG}
 
 This will trigger a [release GitHub action](https://github.com/rancher/cluster-api-provider-rke2/blob/main/.github/workflows/release.yml) that creates a release with RKE2 provider components.
 
+## Tasks
+
+### Prepare main branch for development of the new release
+
+The goal of this task is to bump the versions on the main branch so that the upcoming release version is used for e.g. local development and e2e tests. We also modify tests so that they are testing the previous release.
+
+This comes down to changing occurrences of the old version to the new version, e.g. v1.5 to v1.6, and preparing `metadata.yaml` for a future release version:
+
+#### 1. Update E2E tests
+
+Existing E2E tests that point to a specific version need to be updated to use the new version instead.
+
+1. Add the new release to the list of providers in `test/e2e/config/e2e_conf.yaml` following the format used for previous versions.
```

Comment:
to be specific, this is not adding a new release (simply it is not available yet), but rather changing provider version from i.e `v0.5.99` to `v0.6.99`   - fake upcoming provider version to be tested locally instead of actual GH release